### PR TITLE
Merge main into next

### PR DESCRIPTION
**Pull request by rainest**
**What this PR does / why we need it**:
Merge main into next. Prep for a 2.0.0-alpha.3 release. alpha.2 was about two weeks ago and I'd like to have a released image that has the finalizer changes available, since those matter a lot for permissions stuff.

### CODE REVIEW

**Comment by rainest** [comment]
Minor merge breakage: in main, we bumped some version in the conflicting script, and in next, we renamed said script. Its new location, `railgun/scripts/check-container-environment.sh`, does have the desired version bump. Remind me, how can I resolve that (I just want to delete the file) and still create a clean merge commit from main to next? I don't want to perform the merge in a temp branch and have its name in the history instead.

File: test/integration/test.sh [hunk 1]
```diff
@@ -6,7 +6,7 @@ export CLUSTER_NAME="test-cluster"
 export REGISTRY_NAME="test-local-registry"
 
 export KIND_BINARY=./kind
-export KIND_URL=https://github.com/kubernetes-sigs/kind/releases/download/v0.9.0/kind-linux-amd64
+export KIND_URL=https://github.com/kubernetes-sigs/kind/releases/download/v0.11.1/kind-linux-amd64
```
Comment (ccfishk) [inline test/integration/test.sh:9]: We really need move all kind dependencies into KTF since environment is configured from that repo.

**Review by shaneutt — APPROVED** [review]

**Comment by rainest** [comment]
Action item from @mflendrich: don't use the railgun directory, it's going away. Use top-level singlev1 and singlev2, create a symlink from single to the v2 directory, so that new installs get v2. Probably actually want to defer that to GA (create the two dirs now, but point the symlink to v1 for now, and flip it for FA).

**Comment by ccfishk** [comment]
Still, after more thought - if next branch does not use those new files/modules, the alignment between next and main branch can happen when we merge next back to main branch. Kindly consider about it.

**Comment by shaneutt** [comment]
So as per https://github.com/Kong/kubernetes-ingress-controller/issues/1591, I'm not really able to proactively rebase this one. @rainest any thoughts on how you'd like to proceed on it? Perhaps we can use a 3rd branch as a staging ground?

**Pull request closed without merging**